apps/meteor/app/lib/server/functions/saveUser.ts

```diff
@@ -158,22 +163,20 @@ async function validateUserData(userId, userData) {
 	}
 }
 
-/**
- * Validate permissions to edit user fields
- *
- * @param {string} userId
- * @param {{ _id: string, roles?: string[], username?: string, name?: string, statusText?: string, email?: string, password?: string}} userData
- */
-export async function validateUserEditing(userId, userData) {
+export async function validateUserEditing(userId: IUser['_id'], userData: IUpdateUserParams) {
 	const editingMyself = userData._id && userId === userData._id;
 
 	const canEditOtherUserInfo = await hasPermissionAsync(userId, 'edit-other-user-info');
 	const canEditOtherUserPassword = await hasPermissionAsync(userId, 'edit-other-user-password');
 	const user = await Users.findOneById(userData._id);
 
-	const isEditingUserRoles = (previousRoles, newRoles) =>
+	if (!user) {
```

Comment:
Hmmm I believe this is more of a missing validation than a breaking change. For example, check what we're doing [here](https://github.com/RocketChat/Rocket.Chat/pull/32993/files#diff-b8177d686748dae835de33baf6184906f9a5db359a50b41ab4c672a085ce4c2aR391) **after** `validateUserEditing` ran. We're relying on the `userData._id` field after this function is executed, so we must  assure that the `_id` points to a valid user here

So I think we're just throwing the error earlier here, it shouldn't be breaking. wdyt?

Reply:
I understand your point, but my concern is more about the error message being thrown here, since before nothing was being thrown and now we are throwing... WDYT?

Reply:
> since before nothing was being thrown

hmmm I think that's not exactly true (for all cases). Errors were being thrown before this change, but not **consistently** (I mean, we'd throw an error depending on which fields were being updated)

If `userData._id` is defined, but points to an invalid user, the code will skip [this condition](https://github.com/RocketChat/Rocket.Chat/blob/81629860557f051f7ed1d6d560f8885e587d9c9e/apps/meteor/app/lib/server/functions/saveUser.js#L370). So we'll try to **update** a existing invalid user, and these are the paths we may follow:
1. If `username` or `name` properties are being updated, `saveUserIdentity` [will return false](https://github.com/RocketChat/Rocket.Chat/blob/81629860557f051f7ed1d6d560f8885e587d9c9e/apps/meteor/app/lib/server/functions/saveUserIdentity.ts#L36). Thus, an [error](https://github.com/RocketChat/Rocket.Chat/blob/81629860557f051f7ed1d6d560f8885e587d9c9e/apps/meteor/app/lib/server/functions/saveUser.js#L386) will be thrown
2. If the user tries to set the `email` property, an [error](https://github.com/RocketChat/Rocket.Chat/blob/a40541b61676d23338d72d420995bbc5a211d152/apps/meteor/app/lib/server/functions/setEmail.ts#L54) will also be thrown
3. For bio, nickname, verified, language, settings, roles and status text no error is thrown, indeed. But no (valid) update is performed, also. Still, we trigger all callbacks and events for user update on this case. I believe you agree this is not the intended behavior, but I also believe that's not "worth" of being considered breaking since this is more of an improvement to a case that was being incorrectly handled

wdyt? should we consider it breaking?